### PR TITLE
create a basic demo/editor page for `<verifiable-credential>`

### PR DESCRIPTION
**Pull request by dcrousso**
currently it contains
- a `<textarea>` for the raw JSON `data` of the Verifiable Credential
- a `JSONEditor` to visually create/edit the Credential Manifest
- a `<textarea>` to allow modifying the raw JSON `manifest` of the Credential Manifest

### CODE REVIEW

**Comment by mistermoe** [comment]
https://user-images.githubusercontent.com/4887440/214096439-68edad6e-5703-4c70-8b81-524914df939f.mov

@dcrousso this might be expected behavior, but every time i click "Add item" under `display.title.path` an error is thrown because the path is blank. which technically is expected because blank paths are invalid. just not sure if you're wanting to surface the uncaught exception

Also it wasn't entirely evident that i need to unfocus from the input box for the item to actually be added

**Review by mistermoe — COMMENTED** [review]
This demo seems like it would be helpful for anyone whose already familiar with the [Credential Manifest](https://identity.foundation/credential-manifest/) and [Verifiable Credential](https://www.w3.org/TR/vc-data-model/) specs (which i think is the intent anyway?). Even with a good amount of familiarity with the VC spec but not so much the CM spec, I personally didn't know where to start when i hit the demo page. 

It might be helpful to include a pre-populated example?

**Comment by mistermoe** [comment]
also might be nice to format the VC / CM JSON so that it's easier to read/edit. [ace](https://ace.c9.io/#nav=about) makes that easy IMO. can easily be loaded using cdnjs or jsDelivr

**Comment by dcrousso** [comment]
> @dcrousso this might be expected behavior, but every time i click "Add item" under `display.title.path` an error is thrown because the path is blank. which technically is expected because blank paths are invalid. just not sure if you're wanting to surface the uncaught exception

ah yeah i think that's kinda somewhat expected?  i probably should just have something like
```js
if (!path)
  continue;
```
inside `#getFirstMatchingValue`.  i'll take a deeper look :)

> Also it wasn't entirely evident that i need to unfocus from the input box for the item to actually be added

yeah sadly i think that's a limitation of `JSONEditor`.  i'll look into if there's a way for it to listen for `"input"` instead of `"change"`, but not sure it's possible :/

-----

> This demo seems like it would be helpful for anyone whose already familiar with the [Credential Manifest](https://identity.foundation/credential-manifest/) and [Verifiable Credential](https://www.w3.org/TR/vc-data-model/) specs (which i think is the intent anyway?). Even with a good amount of familiarity with the VC spec but not so much the CM spec, I personally didn't know where to start when i hit the demo page.
> 
> It might be helpful to include a pre-populated example?

yeah I can prepopulate both with something real.  id need something that actually works tho, as the example in <[https://identity.foundation/credential-manifest/]()> doesn't actually point to real images :(

-----

> also might be nice to format the VC / CM JSON so that it's easier to read/edit. [ace](https://ace.c9.io/#nav=about) makes that easy IMO. can easily be loaded using cdnjs or jsDelivr

i coulda sworn i'd made it so that the VC editor would `JSON.stringify(vc, null, 2)`, but maybe i didnt?  im a bit hesitant to load an editor into this cause that's another dependency, but perhaps it's worth it in this case (and frankly at the end of the day this is just a demo page, so we can "afford" some extra dependencies without worry)

**Comment by michaelneale** [comment]
cool - I was wondering if @kirahsapong had any styles/defaults from from SSI admin console we could steal for this?

**Comment by mistermoe** [comment]
Makes sense @dcrousso

**Comment by kirahsapong** [comment]
Nice! 

> cool - I was wondering if @kirahsapong had any styles/defaults from from SSI admin console we could steal for this?
For sure we can get shared styles going.

Also I think the `keydown` listener is preventing keyboard navigation. Allowing `alt/opt` (I think `keycode === 18`) + tab to leave the field would solve it.

**Comment by dcrousso** [comment]
> Also I think the `keydown` listener is preventing keyboard navigation.

I did that intentionally to allow for Tab to actually insert a `\t` instead of switching editors, since some folks (like me) prefer to use tabs instead of spaces for indentation and would find it _really_ annoying to try to edit the JSON only to have it switch focus :(

> Allowing `alt/opt` (I think `keycode === 18`) + tab to leave the field would solve it.

good catch!  i'll only insert the `\t` if not `altKey` (and `ctrlKey` and `metaKey` and `shiftKey`)

File: demo/VerifiableCredential/index.html [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+	previewElement?.remove();
+	previewElement = new VerifiableCredential;
+	try {
+		previewElement.data = JSON.parse(dataRawEditor.value);
+		previewElement.manifest = manifestFormEditor.getValue();
+	} catch { }
+	document.body.appendChild(previewElement);
+
+	if (document.activeElement !== manifestRawEditor)
+		manifestRawEditor.value = JSON.stringify(manifestFormEditor.getValue());
```
Comment (kirahsapong) [inline demo/VerifiableCredential/index.html]: let's format this baddie too?

File: demo/VerifiableCredential/index.html [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+main > section {
+	width: 100%;
+}
+
+input[type="text"] {
+	min-width: 20em;
+}
+
+textarea {
```
Comment (kirahsapong) [inline demo/VerifiableCredential/index.html:46]: Suggesting some styling here for more contrast in our JSON editors, something liiiike
```
background: #000;
color: #fff;
font-size: 1rem;
line-height: 1.67;
padding: 1rem;
```

**Pull request merged**